test/NuGet.CommandLine.Test/RestoreProjectJsonTest.cs

```diff
@@ -9,6 +10,418 @@ namespace NuGet.CommandLine.Test
+            Util.CreateDirectory(repositoryPath);
+            Util.CreateDirectory(Path.Combine(workingPath, ".nuget"));
+            Util.CreateTestPackage("packageA", "1.0.0", repositoryPath, "win8", "packageC", "1.0.0");
+            Util.CreateTestPackage("packageB", "1.0.0-beta", repositoryPath, "win8", "packageC", "2.0.0-beta");
+            Util.CreateTestPackage("packageC", "1.0.0", repositoryPath);
+            Util.CreateTestPackage("packageC", "2.0.0-beta", repositoryPath);
+            Util.CreateConfigForGlobalPackagesFolder(workingPath);
+            Util.CreateFile(workingPath, "project.json",
+                                            @"{
+                                            'dependencies': {
+                                            'packageA': '1.0.0',
+                                            'packageB': '1.0.0-*'
+                                            },
+                                            'frameworks': {
+                                                    'uap10.0': { }
+                                                }
+                                            }");
+
+            string[] args = new string[] {
+                "restore",
+                "-Source",
+                repositoryPath,
+                "-solutionDir",
+                workingPath,
+                "project.json"
+            };
+
+            // Act
+            var r = CommandRunner.Run(
+                nugetexe,
+                workingPath,
+                string.Join(" ", args),
+                waitForExit: true);
+
+            var lockFilePath = Path.Combine(workingPath, "project.lock.json");
+            var lockFileFormat = new LockFileFormat();
+
+            var lockFile = lockFileFormat.Read(lockFilePath);
+
+            var installedC = lockFile.Targets.First().Libraries.Single(package => package.Name == "packageC");
+
+            // Assert
+            Assert.True(0 == r.Item1, r.Item2 + " " + r.Item3);
+            Assert.Equal("2.0.0-beta", installedC.Version.ToNormalizedString());
+        }
+
+        [Fact]
+        public void RestoreProjectJson_RestoreDowngradesStableDependency()
+        {
+            // Arrange
+            var tempPath = Path.GetTempPath();
+            var workingPath = Path.Combine(tempPath, Guid.NewGuid().ToString());
+            var repositoryPath = Path.Combine(workingPath, Guid.NewGuid().ToString());
+            var currentDirectory = Directory.GetCurrentDirectory();
+            var nugetexe = Util.GetNuGetExePath();
+
+            _dirs.TryAdd(workingPath, false);
+
+            Util.CreateDirectory(workingPath);
+            Util.CreateDirectory(repositoryPath);
+            Util.CreateDirectory(Path.Combine(workingPath, ".nuget"));
+            Util.CreateTestPackage("packageA", "1.0.0", repositoryPath, "win8", "packageC", "1.0.0");
+            Util.CreateTestPackage("packageB", "1.0.0", repositoryPath, "win8", "packageC", "[2.1.0]");
```

Comment:
Not sure if I missed it. But, do you want to add a test with a version constraint [4.0.0, ) like in the bug?